### ISSUE TITLE
Mongoose map exceeds call stack

### ISSUE THREAD

**Issue by www-chique** [Bounced-Inc/mongoose-tsgen#28]
I'm using the [Mongoose map example](https://mongoosejs.com/docs/schematypes.html#maps), as follows
```
const EmailTemplateSchema = new Schema(
	{
		content: {
			type: Map,
			of: String,
		},
	}
);
```
Running mtgen shows the following error
![image](https://user-images.githubusercontent.com/29447216/106369684-36a73180-635c-11eb-9188-ff06e1c1df36.png)

However, it works fine if I use Object instead of Map.
Am I doing something wrong like a newbie I am?

Many thanks for your help with all the issues.

**Comment by francescov1** [comment]
This is related to handling Mongoose Maps, my bad! I already had this fix nearly done, just need to finish tests and will release.

**Comment by www-chique** [comment]
@francescov1 - No problem. I'm really grateful to you for this wonderful tool. Take your time.

**Comment by francescov1** [comment]
Published in 7.0.5!

Note that I am aware of a small cleanup that needs to be done to remove duplicate properties on Document types (ie `EmailTemplateDocument["content"] = mongoose.Map<string> & Map<string, string>` even though the Document property should only contain the `mongoose.Map` type). Based on prelim testing it seems like this should be fine for now but I'm not entirely sure if it could cause issues when trying to use a mongoose map (I dont have many Mongoose projects that use maps so I wasn't able to test as much as I would like). Let me know if there are any issues!

**Comment by francescov1** [comment]
> ~~Note that I am aware of a small cleanup that needs to be done to remove duplicate properties on Document types (ie `EmailTemplateDocument["content"] = mongoose.Map<string> & Map<string, string>` even though the Document property should only contain the `mongoose.Map` type). Based on prelim testing it seems like this should be fine for now but I'm not entirely sure if it could cause issues when trying to use a mongoose map (I dont have many Mongoose projects that use maps so I wasn't able to test as much as I would like). Let me know if there are any issues!~~

Addressed in [7.0.6](https://github.com/Bounced-Inc/mongoose-tsgen/pull/30)!

**Comment by www-chique** [comment]
Hey @francescov1 - I ran into the same problem with another Map.
I am using an array of Number as the value of the map.
The error is: "RangeError: Maximum call stack size exceeded".

```
new Schema(
	{
		chapters: {
			type: Map,
			of: [Number],
		},
	},
	{
		timestamps: false,
	}
);
```

Example doc:
```
{
     _id: ".....",
     chapters: {
          "chapter1": [5, 8, 10, 12],
          "chapter2": [1, 2, 20],
     }
}
```
